app/src/main/java/de/danoeh/antennapod/ui/episodeslist/FeedItemMenuHandler.java

```diff
@@ -47,50 +48,88 @@ private FeedItemMenuHandler() {
-        setItemVisibility(menu, R.id.add_to_queue_item, !isInQueue && selectedItem.getMedia() != null);
-        setItemVisibility(menu, R.id.visit_website_item, !selectedItem.getFeed().isLocalFeed()
-                && ShareUtils.hasLinkToShare(selectedItem));
-        setItemVisibility(menu, R.id.share_item, !selectedItem.getFeed().isLocalFeed());
-        setItemVisibility(menu, R.id.remove_inbox_item, selectedItem.isNew());
-        setItemVisibility(menu, R.id.mark_read_item, !selectedItem.isPlayed());
-        setItemVisibility(menu, R.id.mark_unread_item, selectedItem.isPlayed());
-        setItemVisibility(menu, R.id.reset_position, hasMedia && selectedItem.getMedia().getPosition() != 0);
+        if (selectedItems.size() > 1) {
+            canVisitWebsite = false;
+            canShare = false;
+            canShowTranscript = false;
+        }
+
+        setItemVisibility(menu, R.id.skip_episode_item, canSkip);
+        setItemVisibility(menu, R.id.remove_from_queue_item, canRemoveFromQueue);
+        setItemVisibility(menu, R.id.add_to_queue_item, canAddToQueue);
+        setItemVisibility(menu, R.id.visit_website_item, canVisitWebsite);
+        setItemVisibility(menu, R.id.share_item, canShare);
+        setItemVisibility(menu, R.id.remove_inbox_item, canRemoveFromInbox);
+        setItemVisibility(menu, R.id.mark_read_item, canMarkPlayed);
+        setItemVisibility(menu, R.id.mark_unread_item, canMarkUnplayed);
+        setItemVisibility(menu, R.id.reset_position, canResetPosition);
 
         // Display proper strings when item has no media
-        if (hasMedia) {
-            setItemTitle(menu, R.id.mark_read_item, R.string.mark_read_label);
-            setItemTitle(menu, R.id.mark_unread_item, R.string.mark_unread_label);
-        } else {
+        if (selectedItems.size() == 1 && selectedItems.get(0).getMedia() == null) {
             setItemTitle(menu, R.id.mark_read_item, R.string.mark_read_no_media_label);
             setItemTitle(menu, R.id.mark_unread_item, R.string.mark_unread_label_no_media);
+        } else {
+            setItemTitle(menu, R.id.mark_read_item, R.string.mark_read_label);
+            setItemTitle(menu, R.id.mark_unread_item, R.string.mark_unread_label);
         }
 
-        setItemVisibility(menu, R.id.add_to_favorites_item, !isFavorite);
-        setItemVisibility(menu, R.id.remove_from_favorites_item, isFavorite);
-        setItemVisibility(menu, R.id.remove_item, fileDownloaded || isLocalFile);
-        setItemVisibility(menu, R.id.transcript_item, hasTranscript);
+        setItemVisibility(menu, R.id.add_to_favorites_item, canAddFavorite);
+        setItemVisibility(menu, R.id.remove_from_favorites_item, canRemoveFavorite);
+        setItemVisibility(menu, R.id.remove_item, canDelete);
+        setItemVisibility(menu, R.id.download_item, canDownload);
+        setItemVisibility(menu, R.id.transcript_item, canShowTranscript);
```

Comment:
I think this line is causing this bug https://github.com/AntennaPod/AntennaPod/issues/7541